src/adservice/src/main/java/oteldemo/AdService.java

```diff
@@ -194,21 +192,14 @@ public void getAds(AdRequest req, StreamObserver<AdResponse> responseObserver) {
     }
 
     boolean checkAdFailure() {
-      if (featureFlagServiceStub == null) {
-        return false;
-      }
-
+      Client client = OpenFeatureAPI.getInstance().getClient();
+      Boolean boolValue = client.getBooleanValue("adServiceFailure", false);
       // Flip a coin and fail 1/10th of the time if feature flag is enabled
       if (random.nextInt(10) != 1) {
```

Comment:
nitpick, but I'd move this up so you don't fetch a flag value if you don't need to.

Reply:
I changed this up so that we do the fractional eval in flagd rather than in the code.